File: 3.0/python/7.py
import math

A = list(map(int, input().split(":")))
B = list(map(int, input().split(":")))
C = list(map(int, input().split(":")))

asec = A[2] + A[1]*60 + A[0]*60*60
bsec = B[2] + B[1]*60 + B[0]*60*60
csec = C[2] + C[1]*60 + C[0]*60*60

if csec < asec:
    resptime = (24*60*60 + csec - asec) / 2
else:
    resptime = (csec - asec) / 2
    
if resptime % 1 >= 0.5:
    resptime = math.ceil(resptime)
else:
    resptime = math.floor(resptime)
bsec += resptime

hours = bsec // 3600
minuts = (bsec - 3600 * (bsec//3600)) // 60
seconds = (bsec - 3600 * (bsec//3600)) - 60 * minuts
if hours >= 24: hours = hours - 24

print(f"{hours:02}:{minuts:02}:{seconds:02}")

File: 3.0/golang/7.go
package main

import (
	"fmt"
	"math"
)

func main() {
	var ah, am, as int
	var bh, bm, bs int
	var ch, cm, cs int

	fmt.Scanf("%d:%d:%d\n", &ah, &am, &as)
	fmt.Scanf("%d:%d:%d\n", &bh, &bm, &bs)
	fmt.Scanf("%d:%d:%d\n", &ch, &cm, &cs)

	Asec := float64(as + am*60 + ah*60*60)
	Bsec := bs + bm*60 + bh*60*60
	Csec := float64(cs + cm*60 + ch*60*60)

	var resptime float64
	if Csec < Asec {
		resptime = (24*60*60 + Csec - Asec) / 2
	} else {
		resptime = (Csec - Asec) / 2
	}

	resptimeint := int(math.Round(resptime))
	Bsec += resptimeint

	hours := Bsec / 3600
	minuts := (Bsec - 3600*(Bsec/3600)) / 60
	seconds := (Bsec - 3600*(Bsec/3600)) - 60*minuts
	if hours >= 24 {
		hours -= 24
	}

	fmt.Printf("%02d:%02d:%02d\n", hours, minuts, seconds)
}
